python/lsst/obs/hsc/makeTransmissionCurves.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+#
+# Copyright 2017 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+from __future__ import absolute_import, division, print_function
+
+import os
+import glob
+import numpy as np
+
+from lsst.afw.image import TransmissionCurve, Filter
+from lsst.utils import getPackageDir
+from .hscMapper import HscMapper
+
+__all__ = ("getOpticsTransmission", "getSensorTransmission", "getAtmosphereTransmission",
+           "getFilterTransmission",)
+
+DATA_DIR = os.path.join(getPackageDir("obs_subaru"), "hsc", "transmission")
+
+HSC_BEGIN = "2012-12-18"  # initial date for curves valid for entire lifetime of HSC
+
+
+def getLongFilterName(short):
+    """Return a long HSC filter name (e.g. 'HSC-R') that's usable as a data ID
+    value from the short one (e.g. 'r') declared canonical in afw.image.Filter.
+    """
+    HscMapper.addFilters()
+    if short.startswith("HSC") or short.startswith("NB"):
+        return short
+    f = Filter(short)
+    for a in f.getAliases():
+        if a.startswith("HSC") or a.startswith("NB") or a.startswith("IB"):
+            return a
+    return short
+
+
+def readTransmissionCurveFromFile(filename, unit="angstrom", atMin=None, atMax=None):
+    wavelengths, throughput = np.loadtxt(os.path.join(DATA_DIR, filename), usecols=[0, 1], unpack=True)
+    i = np.argsort(wavelengths)
+    wavelengths = wavelengths[i]
+    throughput = throughput[i]
+    if unit == "nm":
+        wavelengths *= 10
```

Comment:
Why choose Angstroms as the standard? I like Angstroms (it's so wonderfully optical-astronomy), but I think that we've deliberately chosen nm in other places (e.g., `lsst.afw.image.Filter`).

Reply:
I didn't really have a strong opinion originall, but angstroms is what I RFC'd back when I started this project, and the afw `TransmissionCurve` classes that document Angstroms as the standard have already been merged.  So I'd rather not backtrack on that now.